contractcourt/channel_arbitrator.go

```diff
@@ -1131,11 +1139,102 @@ func (c *ChannelArbitrator) sweepAnchors(anchors []*lnwallet.AnchorResolution,
 		if err != nil {
 			return err
 		}
+
+		return nil
+	}
+
+	// Calculate the deadline. This is only valid for sweeping the local
+	// anchor.
+	deadline, err := c.findCommitmentDeadline(heightHint)
+	if err != nil {
+		return err
+	}
+
+	for _, anchor := range anchors {
+		// The first item is the anchor of the local commitment tx, we
+		// will use the deadline calcuated from the active HTLCs.
+		if err := sweepWithDeadline(anchor, deadline); err != nil {
+			return err
+		}
+
+		// For the rest anchors, use the default conf target.
+		deadline = uint32(anchorSweepConfTarget)
 	}
 
 	return nil
 }
 
+// findCommitmentDeadline finds the deadline for a commitment transaction by
+// extracting the minimum CLTV from its HTLCs. Currently only the local
+// commitment transaction is suitable to have a confirmation deadline. And it's
+// defined to be the smaller of,
+//   - the least CLTV from outgoing HTLCs, or,
+//   - the least CLTV from incoming HTLCs if the preimage is available.
+func (c *ChannelArbitrator) findCommitmentDeadline(
+	heightHint uint32) (uint32, error) {
+
+	minHeight := uint32(math.MaxUint32)
+
+	// We only use the HTLC set from our local commitment transaction.
```

Comment:
IIUC the HTLC set is the same on all commitments. (I guess they could differ by a single state update)

Reply:
Not necessarily,. we have distinct sets as the HTLCs may differ across the commitments. Consider the case where we force close, but the remote party had a dangling commitment with a newly added HTLC, but they never sent us the signature back.